### PR TITLE
Remove Ruby 2.7.0 warnings

### PR DESCRIPTION
**Pull request by bvicenzo**
This PR implements:
 - Remove URI.escape is obsolete warning;
 - Remove URI. unescape is obsolete warning;
 - Remove 'NOTE: Gem::Specification#rubyforge_project is deprecated...' Warning
 - Fix Breaking tests;
 - Update RSpec and add new syntax;
 - Add Travis Until 2.7.0 Ruby Version;
 - Update Rake gem.

### CODE REVIEW

**Comment by bilby91** [comment]
Any ETA for merging this MR ?

**Comment by coding-bunny** [comment]
Can we have an update on this? The spam from Wasabi in test-suites makes it really hard to get usable output.

File: spec/wasabi/document/authentication_spec.rb [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
+
+      it { is_expected.to eq('http://v1_0.ws.auth.order.example.com/') }
     end
 
     describe '#endpoint' do
-      subject { super().endpoint }
-      it { should == URI("http://example.com/validation/1.0/AuthenticationService") }
+      subject(:endpoint) { document.endpoint }
+
+      it { is_expected.to eq URI('http%3A%2F%2Fexample.com%2Fvalidation%2F1.0%2FAuthenticationService') }
```
Comment (olleolleolle) [inline spec/wasabi/document/authentication_spec.rb:16]: This _seems_ like a change in behavior?

Comment (bvicenzo) [reply spec/wasabi/document/authentication_spec.rb]: @olleolleolle That's true, man...
Seems that the original idea is to transform something like this:
`http%3A%2F%2Fcartoons.com%2Fmgm%2Fold_cartoons.asmx%3Fcartoon%3DTom+%26+Jerry%26year%3D1986`
In this:
`http://cartoons.com/mgm/old_cartoons.asmx?cartoon=Tom%20&%20Jerry&year=1986`

Keeping escaped just query string after the URL...
```rb
url = 'http%3A%2F%2Fcartoons.com%2Fmgm%2Fold_cartoons.asmx%3Fcartoon%3DTom+%26+Jerry%26year%3D1986'
unescaped_url = CGI.unescape(url.to_s)
# => "http://cartoons.com/mgm/old_cartoons.asmx?cartoon=Tom & Jerry&year=1986"
parsed_url = URI(unescaped_url)
% => #<URI::HTTP http://cartoons.com/mgm/old_cartoons.asmx?cartoon=Tom%20&%20Jerry&year=1986>
```
Fixed [here](https://github.com/savonrb/wasabi/pull/75/commits/f6ab0a50900cde84f05c5ac09b82f48a11e54948).
Thanks.

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,9 @@
 sudo: false
 rvm:
-  - 2.0.0
-  - 2.1.8
   - 2.2.2
   - 2.3.0
+  - 2.4.0
+  - 2.5.0
+  - 2.6.0
+  - 2.7.0
```
Comment (olleolleolle) [inline .travis.yml:8]:
```suggestion
  - "2.5"
  - "2.6"
  - "2.7"
```

A nice way to allow rvm to keep the patch versions up-to-date is to rely on its "aliases", example:

https://github.com/rvm/rvm/blob/master/config/known#L12

Comment (olleolleolle) [reply .travis.yml:8]: This Suggestion would match this Savon change, https://github.com/savonrb/savon/commit/b644b513bb7255681631c5ed39a8781404c35135

File: lib/wasabi/parser.rb [hunk 3]
```diff
@@ -117,7 +117,7 @@ def parse_port_type_operations
     end
 
     def parse_operations_parameters
-      root_elements = document.xpath("wsdl:definitions/wsdl:types/*[local-name()='schema']/*[local-name()='element']", 'wsdl' => WSDL).each do |element|
+       document.xpath("wsdl:definitions/wsdl:types/*[local-name()='schema']/*[local-name()='element']", 'wsdl' => WSDL).each do |element|
```
Comment (olleolleolle) [inline lib/wasabi/parser.rb:120]:
```suggestion
      document.xpath("wsdl:definitions/wsdl:types/*[local-name()='schema']/*[local-name()='element']", 'wsdl' => WSDL).each do |element|
```

This whitespace seemed to be wonky.

File: spec/wasabi/document/encoded_endpoint_spec.rb [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-    subject { Wasabi::Document.new fixture(:encoded_endpoint).read }
+  context 'with: encoded_endpoint.wsdl' do
+    let(:document) { Wasabi::Document.new fixture(:encoded_endpoint).read }
 
     describe '#endpoint' do
-      subject { super().endpoint }
-      it { should == URI("http://localhost/soapservice/execute?path=/base/includes/Test+Soap/Return+Rows") }
+      subject(:endpoint) { document.endpoint }
+
+      it { should == URI('http://localhost/soapservice/execute?path=/base/includes/Test%20Soap/Return%20Rows') }
```
Comment (olleolleolle) [inline spec/wasabi/document/encoded_endpoint_spec.rb:10]:
```suggestion
      it { is_expected.to eq(URI('http://localhost/soapservice/execute?path=/base/includes/Test%20Soap/Return%20Rows')) }
```

Perhaps we can use the expect syntax here, too?

**Comment by bvicenzo** [comment]
@olleolleolle Nice! suggestions applied.

File: wasabi.gemspec [hunk 5]
```diff
@@ -12,14 +12,13 @@ Gem::Specification.new do |s|
...
-  s.rubyforge_project = s.name
   s.license = 'MIT'
 
   s.add_dependency "httpi",    "~> 2.0"
   s.add_dependency "nokogiri", ">= 1.4.2"
 
-  s.add_development_dependency "rake",  "~> 0.9"
-  s.add_development_dependency "rspec", "~> 2.14"
+  s.add_development_dependency "rake",  "~> 13.0.1"
+  s.add_development_dependency "rspec", "~> 3.7.0"
```
Comment (olleolleolle) [inline wasabi.gemspec:21]:
```suggestion
  s.add_development_dependency "rake",  "~> 13.0"
  s.add_development_dependency "rspec", "~> 3.7"
```

Is there a reason for being so precise?

**Comment by olleolleolle** [comment]
Cool, thanks for the perseverance!

**Comment by bilby91** [comment]
Any ETA for merging this one ?

File: spec/wasabi/document/authentication_spec.rb [hunk 6]
```diff
@@ -1,37 +1,46 @@
...
-        :authenticate => { :input => "authenticate", :output => "authenticateResponse", :action => "authenticate", :namespace_identifier => "tns" }
-      }
-    end
+      subject(:operations) { document.operations }
+
+      it 'has 1 operation' do
+        expect(operations.size).to eq(1)
+      end
+
+      it 'has authenticate data'do
```
Comment (olleolleolle) [inline spec/wasabi/document/authentication_spec.rb:32]:
```suggestion
      it 'has authenticate data' do
```

File: spec/wasabi/document/multiple_namespaces_spec.rb [hunk 7]
```diff
@@ -1,55 +1,76 @@
...
+      it { is_expected.to eq(:qualified) }
     end
 
     describe '#operations' do
-      subject { super().operations }
-      it do
-      should == { :save => { :input => "Save", :output=>"SaveResponse", :action => "http://example.com/actions.Save", :namespace_identifier => "actions", :parameters => { :article => { :name => "article", :type => "Article" } } } }
+      subject(:operations) { document.operations }
+
+    it 'has 1 operation' do
```
Comment (olleolleolle) [inline spec/wasabi/document/multiple_namespaces_spec.rb:28]: It seems like there's an indentation issue here?

**Comment by bvicenzo** [comment]
@olleolleolle Considerations Done. Thanks.

**Comment by olleolleolle** [comment]
Thanks, @bvicenzo! 

Most of the changes have been integrated into `master` now, and I will close this PR, many of its commits are in the project already.

Thanks to @chaymaeBZ who did the navigating and fixing! 

Many cheers! 🎉

**Comment by bvicenzo** [comment]
@olleolleolle Very thanks.
Please, let me know when we have a new gem version.

**Comment by olleolleolle** [comment]
I can recommend trying it out using a git dependency in your Gemfile - or using the gem "specific_install", which lets you install git versions of gems.

That forms a part of our "community QA", which strengthens upcoming releases.

**Comment by bvicenzo** [comment]
Thanks, I use [dependabot](https://dependabot.com/) to do this...
I follow there so.
Thanks 😄

**Pull request closed without merging**